google/resource_bigtable_instance.go

```diff
@@ -0,0 +1,193 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"cloud.google.com/go/bigtable"
+	"golang.org/x/net/context"
+)
+
+func resourceBigtableInstance() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceBigtableInstanceCreate,
+		Read:   resourceBigtableInstanceRead,
+		Delete: resourceBigtableInstanceDestroy,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"zone": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"display_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+
+			"num_nodes": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      3,
+				ValidateFunc: IntAtLeast(3),
+			},
+
+			"storage_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Default:      "SSD",
+				ValidateFunc: validation.StringInSlice([]string{"SSD", "HDD"}, false),
+			},
+
+			"project": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceBigtableInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	ctx := context.Background()
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	displayName, ok := d.GetOk("display_name")
+	if !ok {
+		displayName = name
+	}
+
+	var storageType bigtable.StorageType
+	switch value := d.Get("storage_type"); value {
+	case "HDD":
+		storageType = bigtable.HDD
+	case "SSD":
+		storageType = bigtable.SSD
+	}
+
+	instanceConf := &bigtable.InstanceConf{
+		InstanceId:  name,
+		DisplayName: displayName.(string),
+		ClusterId:   d.Get("cluster_id").(string),
+		NumNodes:    int32(d.Get("num_nodes").(int)),
+		StorageType: storageType,
+		Zone:        d.Get("zone").(string),
+	}
+
+	c, err := config.bigtableClientFactory.NewInstanceAdminClient(project)
+	if err != nil {
+		return fmt.Errorf("Error starting instance admin client. %s", err)
+	}
+
+	defer c.Close()
```

Comment:
I haven't had Gogland get mad at me, and I don't think we can do anything useful with an error from this - any interesting errors would have come from operations we performed.